### ISSUE TITLE
[Feature] torch.bitwise_where

### ISSUE THREAD

**Issue by zasdfgbnm** [pytorch/pytorch#47954]
## 🚀 Feature

I suggest adding a new operator `torch.bitwise_where(mask, tensor1, tensor2)`

This operator is similar to `torch.where`, except that the operation is done at bit level.

I suggest this because I want to experiment with numerical stability vs the number of mantissa required for a computation, and I want to replace some bits with noise to see what would happen to a computation:

```python
noise = torch.randn(....)
mask = torch.full(..., my_mantissa)
x_with_noise = torch.bitwise_where(mask, x, noise)
y = compute(x_with_noise)
```

**Comment by gchanan** [comment]
can't you do this with existing bitwise operations?

**Comment by zasdfgbnm** [comment]
@gchanan You are right, existing bitwise ops will work. I am just feeling torch.bitwise_where would be slightly more convenient.

**Comment by ezyang** [comment]
I guess... having this method wouldn't be the worst thing?
